terraform/main.mk

```diff
@@ -23,7 +23,23 @@ TFLINT ?= $(DOCKER) run --user "$(CURRENT_USER_ID):$(CURRENT_USERGROUP_ID)" --rm
 TFLOCK ?= $(DOCKER) run --rm --hostname=$(USER)-icmk-terraform -v $(ENV_DIR):/$(ENV_DIR) -v "$(ENV_DIR)/.terraform":/"$(ENV_DIR)/.terraform" -v "$(INFRA_DIR)":"$(INFRA_DIR)" -v $(HOME)/.aws/:/root/.aws:ro -w $(ENV_DIR) -e AWS_PROFILE=$(AWS_PROFILE) -e ENV=$(ENV) hazelops/tflock
 TF_LOG_PATH ?= /$(ENV_DIR)/tflog.txt
 
-TERRAFORM ?= $(DOCKER) run \
+TERRAFORM ?= $(shell echo $$(if [ "$(AWS_MFA_ENABLED)" = "true" ]; then echo "$(TERRAFORM_WITH_MFA)"; else echo "$(TERRAFORM_NO_MFA)"; fi))
+
+TERRAFORM_NO_MFA ?= $(DOCKER) run \
```

Comment:
It looks ugly, but what can you do :(

Maybe you can use GNU Make conditionals to build the list of env vars?

Reply:
@AutomationD made better :)